crates/storage/db/src/tables/mod.rs

```diff
@@ -51,34 +53,111 @@ pub enum TableType {
 /// Number of tables that should be present inside database.
 pub const NUM_TABLES: usize = 25;
 
-/// Default tables that should be present inside database.
-pub const TABLES: [(TableType, &str); NUM_TABLES] = [
-    (TableType::Table, CanonicalHeaders::const_name()),
-    (TableType::Table, HeaderTD::const_name()),
-    (TableType::Table, HeaderNumbers::const_name()),
-    (TableType::Table, Headers::const_name()),
-    (TableType::Table, BlockBodyIndices::const_name()),
-    (TableType::Table, BlockOmmers::const_name()),
-    (TableType::Table, BlockWithdrawals::const_name()),
-    (TableType::Table, TransactionBlock::const_name()),
-    (TableType::Table, Transactions::const_name()),
-    (TableType::Table, TxHashNumber::const_name()),
-    (TableType::Table, Receipts::const_name()),
-    (TableType::Table, PlainAccountState::const_name()),
-    (TableType::DupSort, PlainStorageState::const_name()),
-    (TableType::Table, Bytecodes::const_name()),
-    (TableType::Table, AccountHistory::const_name()),
-    (TableType::Table, StorageHistory::const_name()),
-    (TableType::DupSort, AccountChangeSet::const_name()),
-    (TableType::DupSort, StorageChangeSet::const_name()),
-    (TableType::Table, HashedAccount::const_name()),
-    (TableType::DupSort, HashedStorage::const_name()),
-    (TableType::Table, AccountsTrie::const_name()),
-    (TableType::DupSort, StoragesTrie::const_name()),
-    (TableType::Table, TxSenders::const_name()),
-    (TableType::Table, SyncStage::const_name()),
-    (TableType::Table, SyncStageProgress::const_name()),
-];
+/// The table visitor
+pub trait TableViewer<R> {
+    /// Type of error to return
+    type Error;
+
+    /// Allows to operate on specific table type
+    fn view<T: Table>(&self) -> Result<R, Self::Error>;
+}
```

Comment:
Can we update the docs here to make it more clear what it does? The doc currently references it as a table visitor, it is not entirely clear what the docs for `view` implies either

Looking at the code I sort of get what it is used for, but some clearer docs would be good :)